### DISCUSSION TITLE
Formulaire différent selon la boutique utilisée

### DISCUSSION THREAD

**Discussion by francoistomasi** [PayGreen/documentation#11] [Q&A]
Bonjour,

Durant l'intégration d'un formulaire dédié à l'enregistrement de cartes bancaires, j'ai effectué plusieurs tests avec 2 boutiques sandbox.
Je me suis rendu compte que les champs injectés par le script PayGreenJS n'étaient pas les mêmes.

![Shop #1](https://github.com/PayGreen/documentation/assets/6708698/bb547309-2fb3-486a-9d80-158b751397f2)
![Shop #2](https://github.com/PayGreen/documentation/assets/6708698/69018d0a-322e-4ea0-9bf1-882094bf923a)

Dans le cas de la première photo (Shop #1), la zone de clic des champs dépasse le bloc parent et empêche le clic sur les autres champs.

![Problem on Shop #1](https://github.com/PayGreen/documentation/assets/6708698/43b971c2-f131-46a6-9025-6cd72e986fa8)

Y-a-t-il différentes versions de la librairie en test ? Est-t-il possible d'avoir la même version ou alors un correctif sur la version du Shop #1 ?

Merci d'avance,

**Comment by enzobillisPG** [comment]
Bonjour Francois,

Il n'y a qu'une seule version de la librairie il s'agit de la latest. La différence d'affichage des champs que vous rencontrez est due aux payment configs utilisés à l'initialisation de PGJS.

Dans le premier cas, il s'agit d'un payment config bank_card en SVAD. Dans le deuxième cas il s'agit d'un bank_card VAD. Dans le cadre des paiements par carte bancaire SVAD nous faisons appel à la librairie de notre prestataire et affichons ses champs embarqués à la place de ceux de Paygreen. Ceci explique la différence de style. Une uniformisation est en cours développement de notre côté.

Concernant l'overlap du deuxième cas il s'agit des champs de notre partenaire. Pouvez-vous me fournir une capture d'écran avec le devtool ouvert et le noeud sélectionné ? De plus, il me faudrait le navigateur utilisé, l'OS, ainsi que votre taille d'écran.

Néanmoins si vous rencontrez un problème d'affichage sur nos champs autre que cette différence n'hésitez pas à nous en faire part.

Bonne journée,
Enzo

**Reply by francoistomasi** [reply]
Bonjour,

Merci pour votre retour.

J'ai essayé de modifier le type de contrat SVAD/VAD mais impossible de modifier ce paramètre depuis le backoffice Paygreen. (cf. Screenshots).
L'affichage n'est d'ailleurs pas le même pour les 2 boutiques 🤔

![Shop #1](https://github.com/PayGreen/documentation/assets/6708698/032e4de5-6631-4728-8123-0098bea3d69b)
![Shop #2](https://github.com/PayGreen/documentation/assets/6708698/1d232457-c1b9-4a51-a2c7-2e23dfdcaf46)

Voici la capture d'écran du nœud sélectionné, le problème est présent sur les 3 inputs générés ayant la classe `field`

![Input Node](https://github.com/PayGreen/documentation/assets/6708698/a8a6e9d0-9ead-4173-a482-53622840d8ed)

Ce test à été effectué sur MacOS 13.3.1 - Google Chrome 114.0.5735.106
La résolution du navigateur est de 1080x1200, sachant que le problème est présent à n'importe quel taille.

Merci d'avance,

**Comment by enzobillisPG** [comment]
Bonjour,
Merci pour votre réponse détaillée. Pour passer depuis le back-office un payment config bank_card en SVAD vous devez sélectionner "Je n'ai pas de contrat de vente à distance" (VAD signifiant vente à distance) et sauvegarder. 

Pour modifier cette configuration je vous invite à supprimer et recréer le payment config.

Concernant votre probleme d'affichage, je n'arrive pas à reproduire de mon côté. Je pense que le probleme vient du style appliqué dans les containers au dessus. Avez vous modifié le style de base de PGJS ?

![image](https://github.com/PayGreen/documentation/assets/91066167/e025a55a-f45b-4618-a487-40c2a0f7692b)

Enzo
